### PR TITLE
[TEVA 1025] accordian group used for filters

### PR DESCRIPTION
**Pull request by alexbowen**
a component that can be used for GDS accordian groups (MAT)

https://dfedigital.atlassian.net/browse/TEVA-1025

## Changes in this PR:

- this also changes the JS folder structure to align with GDS and be a little more intuitive
Screenshot 2020-07-08 at 12 03 54
Screenshot 2020-07-08 at 12 03 46
Screenshot 2020-07-08 at 12 03 34

### CODE REVIEW

File: app/frontend/styles/application.scss [hunk 1]
```diff
@@ -16,8 +16,9 @@ $govuk-image-url-function: frontend-image-url;
 @import 'base/mixins';
 @import 'base/utilities';
 @import 'components/awaiting_feedback_vacancies';
-@import '../src/lib/autocomplete';
+@import '../src/lib/ui/patterns/autocomplete';
 @import 'components/banners';
+@import 'components/accordian-group';
```
Comment (david-mears-dfe) [inline app/frontend/styles/application.scss:21]: accordian -> accordion

Comment (david-mears-dfe) [reply app/frontend/styles/application.scss:21]: This applies to code and to filenames

File: app/views/vacancies/index.html.haml [hunk 2]
```diff
@@ -23,19 +23,20 @@
             = render partial: 'vacancy', locals: { vacancy: vacancy }
```
Comment (david-mears-dfe) [inline app/views/vacancies/index.html.haml:23]: What are the changes to this view?

File: app/frontend/styles/application.scss (hunk 1 above)
Comment (alexbowen) [reply app/frontend/styles/application.scss]: just pushed to correct this everywhere is wrong

File: app/views/vacancies/index.html.haml (hunk 2 above)
Comment (alexbowen) [reply app/views/vacancies/index.html.haml:23]: hmm im not sure there should be any, am checking

Comment (alexbowen) [reply app/views/vacancies/index.html.haml]: reverted whatever happened here, wasnt intentional

File: app/views/vacancies/index.html.haml [hunk 3]
```diff
@@ -45,4 +45,4 @@
     .pagination-and-stats
       .pagination-results
         = paginate @vacancies_search.vacancies
-      %p.govuk-body#vacancies-stats-bottom{ 'aria-label': 'Number of results' }
+      %p.govuk-body#vacancies-stats-bottom{ 'aria-label': 'Number of results' }
```
Comment (cpjmcquillan) [inline app/views/vacancies/index.html.haml:48]: Newline

File: app/frontend/styles/components/accordion-group.scss [hunk 4]
```diff
@@ -0,0 +1,41 @@
+.js-enabled .govuk-accordion {
+    border: none;
+    margin-bottom: 10px;
+}
+
+.accordion-group {
+    .accordion-content__group.govuk-accordion__section--expanded {
+        border-bottom:1px solid $govuk-border-colour;
```
Comment (cpjmcquillan) [inline app/frontend/styles/components/accordion-group.scss:8]: missing space

File: app/frontend/styles/components/accordion-group.scss [hunk 5]
```diff
@@ -0,0 +1,41 @@
+.js-enabled .govuk-accordion {
+    border: none;
+    margin-bottom: 10px;
```
Comment (cpjmcquillan) [inline app/frontend/styles/components/accordion-group.scss:3]: Could we use `margin-bottom: govuk-spacing(X)`

File: app/frontend/styles/components/accordion-group.scss [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+.accordion-group {
+    .accordion-content__group.govuk-accordion__section--expanded {
+        border-bottom:1px solid $govuk-border-colour;
+        padding-bottom: 10px;
```
Comment (cpjmcquillan) [inline app/frontend/styles/components/accordion-group.scss:9]: Same here, could we use `padding-bottom: govuk-spacing(X)`

File: app/views/shared/_accordion_group.html.haml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+-# Usage to include accordion group
+-# = render partial: 'shared/accordion_group', locals: { :title => 'Job Filters', items: [{ :title => 'locations' }, { :title => 'created by' }] }
+
+.accordion-group.govuk-accordion{ id: 'accordion-default', data: { module: "govuk-accordion"} }
+  - if title.present?
+    .govuk-accordion__section.govuk-accordion__title
+      .govuk-accordion__section-header
+        %h3.govuk-accordion__section-heading.accordion-group-heading
+          %span.govuk-accordion__section-button{ id: "accordion-default-heading-title" }
```
Comment (cpjmcquillan) [inline app/views/shared/_accordion_group.html.haml:9]: You can add the id like `#accordion-default-heading-title` if you prefer

File: app/views/shared/_accordion_group.html.haml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  - items.each_with_index do |filter, index|
+    .govuk-accordion__section.accordion-content__group
+      .govuk-accordion__section-header
+        %h3.govuk-accordion__section-heading
+          %span.govuk-accordion__section-button{ id: "accordion-default-heading-#{index}" }
+            = filter[:title]
+
+      .govuk-accordion__section-content{ id: "accordion-default-content-#{index}", aria: { labelledby: "accordion-default-heading-#{index}" } }
+        %p.govuk-body
+          = filter[:title] + ' content'
```
Comment (cpjmcquillan) [inline app/views/shared/_accordion_group.html.haml:25]: I think it's ok to dictate that the hash passed in the items array contains `content: something` and then just have `= filter[:content]` here

File: app/views/shared/_accordion_group.html.haml [hunk 9]
```diff
@@ -0,0 +1,25 @@
+-# Usage to include accordion group
+-# = render partial: 'shared/accordion_group', locals: { :title => 'Job Filters', items: [{ :title => 'locations' }, { :title => 'created by' }] }
```
Comment (cpjmcquillan) [inline app/views/shared/_accordion_group.html.haml:2]: `render '/shared/accordion_group', title: 'Job filters', items: [{title: 'Locations', content: some_content_or_form_field}]`should work too

**Comment by alexbowen** [comment]
included in PR for full feature

**Pull request closed without merging**